meterfs-migrate/migrate.c

```diff
@@ -0,0 +1,673 @@
+/*
+ * Phoenix-RTOS
+ *
+ * meterfs migration tool
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * %LICENSE%
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define MAX_FILES         64
+#define SIZE_SECTOR       4096
+#define HGRAIN            32 /* Must be able divide sector size */
+#define HEADER_SECTOR_CNT 2
+
+
+typedef struct {
+	unsigned int nvalid:1;
+	unsigned int no:31;
+} __attribute__((packed)) index_t;
+
+
+typedef struct {
+	unsigned int sector;
+	uint32_t sectorcnt;
+	uint32_t filesz;
+	uint32_t recordsz;
+	char name[8];
+} __attribute__((packed)) fileheader_t;
+
+
+typedef struct {
+	index_t id;
+	uint32_t filecnt;
+	uint32_t checksum;
+	unsigned char magic[4];
+} __attribute__((packed)) header_t;
+
+
+typedef struct {
+	index_t id;
+	uint32_t checksum;
+	unsigned char data[];
+} __attribute__((packed)) entry_t;
+
+
+typedef struct {
+	index_t id;
+	uint32_t filecnt;
+	unsigned char magic[4];
+} __attribute__((packed)) header_old_t;
+
+
+typedef struct {
+	index_t id;
+	unsigned char data[];
+} __attribute__((packed)) entry_old_t;
+
+
+static const unsigned char magicConst[4] = { 0x66, 0x41, 0x4b, 0xbb };
+static const unsigned char oldMagicConst[4] = { 0xaa, 0x41, 0x4b, 0x55 };
+
+
+static struct {
+	unsigned char buff[SIZE_SECTOR];
+	unsigned int freeSector;
+	long partOffset;
+} common;
```

Comment:
According to our convention:
`meterfs_common`

Reply:
It's not meterfs, it's meterfs-migrate. Also, this is a stand-alone component, so tbh I don't see a point in making use of prefix for all names here.